scenes/quests/lore_quests/quest_001/1_music_puzzle/components/music_puzzle.gd

```diff
@@ -26,6 +26,12 @@ var _is_demo: bool = false
 func _ready() -> void:
 	if not Engine.is_editor_hint():
 		xylophone.note_played.connect(_on_note_played)
+		
+	for i in fires.size():
+		if fires[i].is_ignited:
+			_current_melody = i + 1
+			_position = 0
+			break
```

Comment:
If there are 4 fires and the first 2 are ignited, this `break` is wrong.

Reply:
Fixed!